src/controllers/userController.js

```diff
@@ -1,43 +1,105 @@
-                if (token){
-                    response.status(201).json({'accessToken': token, Message: 'User created'});
-                }
-                else {
-                    ApplicationError.internalServerError(`An error occured failed`,response);
-                }
-            }
-       } else {
-           ApplicationError.validationError(validate.error.details[0].context.label,response);
-       }
-    } catch (error) {
-        ApplicationError.internalServerError(`${error}`,response);
-        next(error);    
+        const user = await userService.addUser(userData);
+        const salt = await bcrypt.genSalt(10);
+        user.password = await bcrypt.hash(user.password, salt);
+        await user.save();
+        const token = await tokenGenerator.generateAccessToken({
+          email: user.email,
+          id: user.id
+        });
+        if (token) {
+          response
+            .status(201)
+            .json({ accessToken: token, Message: 'User created' });
+        } else {
+          ApplicationError.internalServerError(
+            `An error occured failed`,
+            response
+          );
+        }
+      }
+    } else {
+      ApplicationError.validationError(
+        validate.error.details[0].context.label,
+        response
+      );
     }
-}
-export default {registerNew}
\ No newline at end of file
+  } catch (error) {
+    ApplicationError.internalServerError(`${error}`, response);
+    next(error);
+  }
+};
+export const login = async (req, res) => {
+  try {
+    const user = await Users.findOne({ where: { email: req.body.email } });
+
+    if (!user) {
+      return res.status(400).json({ error: 'This email does not exist' });
+    }
+    if (!user.isVerified) {
+      return res.status(400).json({ error: 'Verify to log into your account' });
+    }
+    const verifyPassword = await bcrypt.compare(
+      req.body.password,
+      user.password
+    );
+    if (!verifyPassword) {
+      return res.status(400).json({ error: 'Invalid Password' });
+    }
+    const token = await storeToken(user);
+    return res.status(200).json({
+      ...token,
+      message: 'User logged in successfully'
+    });
+  } catch (error) {
+    return res.status(400).json({
+      error: 'Invalid input'
+    });
+  }
+};
+export const refreshToken = async (req, res) => {
+  try {
+    const {refreshTokenKey} = req.body;
+
+    if (!refreshTokenKey) {
+      return res.status(400).json({ status: 400, message: 'Bad request' });
+    }
+    const token = await storeToken(null, refreshTokenKey);
+    if (!token) {
+      return res
+        .status(400)
+        .json({ status: 400, message: 'Invalid refresh token' });
+    }
+    return res.status(200).json({
+      ...token,
+      status: 200,
+      message: 'Access token created successfully'
+    });
+  } catch (error) {
+    return error.message;
+  }
+};
```

Comment:
There is a helper function for this in utils/helpers please rebase and use it , remember to use user id and email in object to get a token instead of passing the entire user object , not all values are required and can be retrieved based on id and email of the user as they are unique and one is primary key